### PR TITLE
Styling fixes

### PR DESCRIPTION
**Pull request by IljaDaderko**
- Addresses some styling related issues in #585 (tackled tasks are ticked off)
- Added accessor function for ethereumClient (without it accessing its state resulted in outdated values)
- Updated deps
- Extended toast component to be used for error messages

### CODE REVIEW

File: packages/ui/src/partials/w3m-mobile-wallet-selection/styles.ts [hunk 1]
```diff
@@ -16,7 +17,7 @@ export default css`
 
   w3m-wallet-button,
   w3m-view-all-wallets-button {
-    padding: 0 10px;
+    padding: 0 calc((100% - 255px) / 11);
```
Comment (devceline) [inline packages/ui/src/partials/w3m-mobile-wallet-selection/styles.ts:20]: This seems a bit too "magic number"y. Anyway we can use percentages or add comments explaining the reasoning behind the values

File: packages/ui/src/components/w3m-modal-header/styles.ts [hunk 2]
```diff
@@ -10,17 +10,20 @@ export default css`
...
-  .w3m-back-btn:hover {
-    opacity: 0.6;
+  .w3m-modal-header button:hover {
+    opacity: 0.5;
+  }
+
+  .w3m-back-btn {
+    position: absolute;
+    left: 0;
+    margin-left: -17px;
```
Comment (devceline) [inline packages/ui/src/components/w3m-modal-header/styles.ts:26]: Replacing `-17px` with `-5%` results in a nearly identical look but it's a lot more clear and perhaps will be more responsive going forward.

**Review by devceline — APPROVED** [review]
LGTM

**Review by crypblizz8 — APPROVED** [review]
LGTM 🤝

Checked:
- Rework on ethereal client hooks
- Smoother transition on mobile
- Error toast works great 
- Updated packages.

Small Check:
- Want to double check `600px` is the breakpoint we set since it was `450px` before. Assumption is that `450` was too small and most mobile is around `576px` these days. 

Testing Process: 
- Read through code over PR 
- Test the edge cases locally
- Tested ErrorToast on `w3m-metamask-connector-view` / `w3m-walletconnect-connector-view`

File: packages/ui/src/partials/w3m-mobile-wallet-selection/styles.ts (hunk 1 above)
Comment (IljaDaderko) [reply packages/ui/src/partials/w3m-mobile-wallet-selection/styles.ts:21]: You are right, I added a comment explaining what this does / is used for. Was not able to achieve it with just percentages since actual wallet icons / buttons have fixed widths, this this calculation.

File: packages/ui/src/components/w3m-modal-header/styles.ts (hunk 2 above)
Comment (IljaDaderko) [reply packages/ui/src/components/w3m-modal-header/styles.ts:26]: Most of ui elements in web3modal are fixed, including most paddings, so it is fine here. Main responsiveness in the modal comes from things like `display: flex` with `justify-content: space-between` for example.

Comment (devceline) [reply packages/ui/src/components/w3m-modal-header/styles.ts:26]: Got it

**Comment by IljaDaderko** [comment]
@crypblizz8 

> Want to double check 600px is the breakpoint we set since it was 450px before. Assumption is that 450 was too small and > most mobile is around 576px these days.

yep exactly, this is following Maciejs feedback when we tested on an iPhone Plus sized phone.
